OpenMTR/FileHandler.cs

```diff
@@ -0,0 +1,267 @@
+        }
+
+
+
+        /// <summary>
+        /// Replace forward slashes with backslashes and add a trailing backslash if not present
+        /// </summary>
+        /// <param name="input"></param>
+        /// <returns>A standardized path</returns>
+        public static string standardizePath(string input)
+        {
+            string output = input.Replace('/','\\').Trim();
+
+            if (output[output.Length - 1] == '\\') return output;
+            return output + "\\";
+        }
+
+
+
+        /// <summary>
+        /// Returns the integer contained in a plain text file
+        /// </summary>
+        /// <param name="fileName">The name of file with complete path</param>
+        /// <returns>Whatever integer was found in the file</returns>
+        public static int ConvertToInt(string fileName, bool noFail)
+        {
+            string number = File.ReadAllText(fileName);
+
+            number = number.Replace("[\n\r]", "").Trim(); // I was going to trim anyway, so I might as well get rid of new lines while I'm at it
+
+            if (noFail) // If the file contains some kind of comment or other silliness this can be used to purge every non-numeric character
+            {
+                number = "0" + number.Replace("[^0-9]","");
+            }
+
+            return int.Parse(number);
+        }
+        public static int ConvertToInt(string fileName)
+        {
+            return ConvertToInt(fileName, false);
+        }
+
+
+
+        //=================================== PUBLIC METHODS ==============================================
+
+        
+        
+        /// <summary>
+        /// Check if there is a next file to move on to
+        /// </summary>
+        /// <returns></returns>
+        public bool HasNext()
+        {
+            return pos < filenames.Length;
+        }
+
+
+        /// <summary>
+        /// Moves to the next file and returns it and its associated metadata in a struct
+        /// </summary>
+        /// <returns>struct containing a bitmap and an integer</returns>
+        public Files Next()
+        {
+            pos++;
+            while (!Regex.IsMatch(ImageFile, extensions) && HasNext()) pos++;
+            return processFiles();
+        }
+
+
+
+        //=================================== STATIC METHODS WITH PRIVATE USES =============================
+
+
+
+        /// <summary>
+        /// Process two files and return a bitmap and an integer contained in a struct
+        /// </summary>
+        /// <param name="imageFile">The file to be converted to a bitmap</param>
+        /// <param name="metadataFile">The file to be converted to an integer</param>
+        /// <param name="ignoreFNF">If there's no metadata file enter -1 instead of throwing a FileNotFoundException</param>
+        /// <returns>A struct containing the bitmap and integer</returns>
+        public static Files processFiles(string imageFile, string metadataFile, bool ignoreFNF = false)

```

Comment:
I will modify variable names, sorry about that.
An FNF could (in theory) occur if metadata was missing, or if an image file was missing from the list. I wanted a way of easily sorting between the two and catching an FNF exception here, I thought, would be a decent way of handling sorting between the two. A -1 value essentially does what you described because it can be easily filtered for, no value should be negative. (Now I come to think of it, that's wrong, somebody could actually have solar or some collector that runs the meter backward, this requires extra thought.